--- src/guake.py.orig	2011-09-02 15:50:50.000000000 -0400
+++ src/guake.py	2011-09-02 16:22:18.000000000 -0400
@@ -571,6 +571,11 @@
         # reset to avoid problems of repeated tab names.
         self.tab_counter = 0
 
+        # Instead of blindly incrementing the tab counter, recycle
+        # the tab numbers. "Terminal 125123" looks annoying !!!
+        self.available_tab_numbers = [] # Contains tab_number values
+        self.registered_tab_numbers = [] # Contains (PID, tab_number) tuples
+
         # holds fullscreen status
         self.fullscreen = False
 
@@ -1115,6 +1120,32 @@
                 os.environ['https_proxy'] = 'http://%s:%d' % (
                     ssl_host, ssl_port)
 
+    def request_tab_number(self, pid):
+        """Returns the next available tab number
+           Try to pick a number from available_tab_numbers
+           If the list is empty, use the tab_counter value along with
+             the PID of the terminal to register it. Increment tab counter
+        """
+        if self.available_tab_numbers:
+            self.available_tab_numbers.sort()
+            tab_number = self.available_tab_numbers.pop(0)
+            self.registered_tab_numbers.append( (pid, tab_number) )
+            return tab_number
+        else:
+            tab_number = self.tab_counter
+            self.registered_tab_numbers.append( (pid, tab_number) )
+            self.tab_counter += 1
+            return tab_number
+
+    def release_tab_number(self, pid):
+        """Move the tab number with the corresponding PID to
+           available tab numbers list
+        """
+        for pid_tab_pair in self.registered_tab_numbers:
+            if pid_tab_pair[0] == pid:
+                self.available_tab_numbers.append(pid_tab_pair[1])
+                self.registered_tab_numbers.remove(pid_tab_pair)
+
     def add_tab(self, directory=None):
         """Adds a new tab to the terminal notebook.
         """
@@ -1139,7 +1170,7 @@
         self.pid_list.append(pid)
 
         # Adding a new radio button to the tabbar
-        label = _('Terminal %s') % self.tab_counter
+        label = _('Terminal %s') % str(self.request_tab_number(pid))
         tabs = self.tabs.get_children()
         parent = tabs and tabs[0] or None
 
@@ -1155,7 +1186,6 @@
         bnt.show()
 
         self.tabs.pack_start(bnt, expand=False, padding=1)
-        self.tab_counter += 1
 
         self.notebook.append_page(box, None)
         self.notebook.set_current_page(self.notebook.page_num(box))
@@ -1171,6 +1201,8 @@
         self.term_list.pop(pagepos).destroy()
         pid = self.pid_list.pop(pagepos)
 
+        self.release_tab_number(pid)
+
         if kill:
             start_new_thread(self.delete_shell, (pid,))
 
