--- a/machine.h
+++ b/machine.h
@@ -220,12 +220,6 @@
     incr/decr operation specifiers */
 #define MD_MAX_MASK		2048
 
-/* internal decoder state */
-extern enum md_opcode md_mask2op[];
-extern unsigned int md_opoffset[];
-extern unsigned int md_opmask[];
-extern unsigned int md_opshift[];
-
 /* global opcode names, these are returned by the decoder (MD_OP_ENUM()) */
 enum md_opcode {
   OP_NA = 0,	/* NA */
@@ -236,6 +230,12 @@
   OP_MAX	/* number of opcodes + NA */
 };
 
+/* internal decoder state */
+extern enum md_opcode md_mask2op[MD_MAX_MASK + 1];
+extern unsigned int md_opoffset[OP_MAX];
+extern unsigned int md_opmask[OP_MAX];
+extern unsigned int md_opshift[OP_MAX];
+
 /* enum md_opcode -> description string */
 #define MD_OP_NAME(OP)		(md_op2name[OP])
 extern char *md_op2name[];
