--- src/disper.py.orig	2012-02-13 06:57:18.800000006 -0500
+++ src/disper.py	2012-02-13 07:56:38.440000008 -0500
@@ -114,6 +114,13 @@
             help='cycle through the list of cycle stages')
         self.parser.add_option_group(group)
 
+        # Handle help and version options explicitly
+        if "-h" in sys.argv or "--help" in sys.argv:
+            self.parser.print_help()
+            sys.exit(0)
+        if "--version" in sys.argv:
+            self.parser.print_version()
+            sys.exit(0)
 
     def add_option(self, *args, **kwargs):
         '''adds an option to the parser. Implements append_const for Python<2.5 too'''
