--- ipc/ipc_platform_file.cc.orig	2012-01-24 01:13:02.850000001 -0500
+++ ipc/ipc_platform_file.cc	2012-01-24 01:13:58.660000001 -0500
@@ -4,6 +4,10 @@
 
 #include "ipc/ipc_platform_file.h"
 
+#if defined(OS_POSIX)
+#include <unistd.h>
+#endif
+
 namespace IPC {
 
 PlatformFileForTransit GetFileHandleForProcess(base::PlatformFile handle,
--- ipc/file_descriptor_set_posix.cc.orig	2012-01-24 01:03:41.490000002 -0500
+++ ipc/file_descriptor_set_posix.cc	2012-01-24 01:04:49.360000001 -0500
@@ -4,6 +4,7 @@
 
 #include "ipc/file_descriptor_set_posix.h"
 
+#include <unistd.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 
--- ipc/ipc_channel.h.orig	2012-01-24 00:57:03.750000002 -0500
+++ ipc/ipc_channel.h	2012-01-24 01:00:46.320000002 -0500
@@ -10,6 +10,10 @@
 #include "ipc/ipc_channel_handle.h"
 #include "ipc/ipc_message.h"
 
+#if defined(OS_POSIX)
+#include <sys/types.h>
+#endif
+
 namespace IPC {
 
 //------------------------------------------------------------------------------
--- seccompsandbox/library.h.orig	2012-01-24 00:55:34.790000002 -0500
+++ seccompsandbox/library.h	2012-01-24 00:55:45.030000002 -0500
@@ -12,6 +12,7 @@
 #include <string>
 #include <string.h>
 #include <sys/mman.h>
+#include <sys/types.h>
 
 #include "maps.h"
 
--- base/message_pump_libevent.cc.orig	2012-01-24 11:17:49.730000002 -0500
+++ base/message_pump_libevent.cc	2012-01-24 11:18:26.800000002 -0500
@@ -4,6 +4,7 @@
 
 #include "base/message_pump_libevent.h"
 
+#include <unistd.h>
 #include <errno.h>
 #include <fcntl.h>
 
--- chrome/browser/policy/policy_path_parser_linux.cc.orig	2012-01-24 17:10:53.974386281 -0500
+++ chrome/browser/policy/policy_path_parser_linux.cc	2012-01-24 17:11:06.254386282 -0500
@@ -8,6 +8,8 @@
 
 #include "base/logging.h"
 
+#include <unistd.h>
+
 namespace policy {
 
 namespace path_parser {
--- chrome/browser/diagnostics/diagnostics_main.cc.orig	2012-01-24 16:47:48.744386282 -0500
+++ chrome/browser/diagnostics/diagnostics_main.cc	2012-01-24 16:58:42.744386282 -0500
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include "build/build_config.h"
 #include "chrome/browser/diagnostics/diagnostics_main.h"
 
 #if defined(OS_POSIX)
--- chrome/browser/diagnostics/recon_diagnostics.cc.orig	2012-01-24 16:47:48.744386282 -0500
+++ chrome/browser/diagnostics/recon_diagnostics.cc	2012-01-24 16:58:49.494386282 -0500
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include "build/build_config.h"
 #include "chrome/browser/diagnostics/recon_diagnostics.h"
 
 #include <string>
--- chrome/browser/diagnostics/diagnostics_model_unittest.cc.orig	2012-01-24 16:47:48.744386282 -0500
+++ chrome/browser/diagnostics/diagnostics_model_unittest.cc	2012-01-24 16:58:51.924386282 -0500
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include "build/build_config.h"
 #include "chrome/browser/diagnostics/diagnostics_model.h"
 
 #include "base/command_line.h"
--- chrome/browser/diagnostics/diagnostics_model.cc.orig	2012-01-24 16:47:48.744386282 -0500
+++ chrome/browser/diagnostics/diagnostics_model.cc	2012-01-24 16:58:54.464386282 -0500
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include "build/build_config.h"
 #include "chrome/browser/diagnostics/diagnostics_model.h"
 
 #include <algorithm>
