--- src/eix-diff.cc.orig	2012-01-03 23:14:54.000000000 -0500
+++ src/eix-diff.cc	2012-01-03 23:15:09.000000000 -0500
@@ -41,6 +41,8 @@
 #include <cstdio>
 #include <cstdlib>
 
+#include <unistd.h>
+
 #define VAR_DB_PKG "/var/db/pkg/"
 
 using namespace std;
--- src/eix.cc.orig	2012-01-03 23:15:23.000000000 -0500
+++ src/eix.cc	2012-01-03 23:15:30.000000000 -0500
@@ -47,6 +47,8 @@
 #include <cstdlib>
 #include <cstring>
 
+#include <unistd.h>
+
 #define VAR_DB_PKG "/var/db/pkg/"
 
 using namespace std;
--- src/eixTk/md5.cc.orig	2012-01-03 23:15:50.000000000 -0500
+++ src/eixTk/md5.cc	2012-01-03 23:16:16.000000000 -0500
@@ -15,6 +15,8 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 
+#include <unistd.h>
+
 using namespace std;
 
 inline static uint32_t
